backend/src/core/user/port/user-repository.ts

```diff
@@ -1,8 +1,11 @@
 import { User } from '@prisma/client';
+import { UploadApiResponse } from 'cloudinary';
 import { UserSignUpRequestDto, UserSignUpResponseDto } from 'shared/build';
 
 export interface UserRepository {
   getAll(): Promise<User[]>;
 
   createUser(userRequestDto: UserSignUpRequestDto): Promise<UserSignUpResponseDto>;
+
+  uploadAvatar(base64Str: string): Promise<UploadApiResponse>;
```

Comment:
You should not mix a repository for working with a database and a service (adapter) for working with the cloud for storing image files. It should be separate files.

Reply:
In your case should directly inject in UserService the CloudinaryAdapter that you already created.